openedx/core/djangoapps/coursegraph/management/commands/dump_to_neo4j.py

```diff
@@ -168,44 +164,91 @@ def add_to_transaction(neo4j_entities, transaction):
             transaction.create(entity)
 
     @staticmethod
-    def should_dump_course(course_key):
+    def get_command_last_run(course_key, graph):
+        """
+        This information is stored on the course node of a course in neo4j
+        Args:
+            course_key: a CourseKey
+            graph: a py2neo Graph
+
+        Returns: None, if there's no record of this command last being run, or
+            the datetime that the command was last run, converted into text.
+
+        """
+        selector = NodeSelector(graph)
+        course_node = selector.select(
+            "course",
+            course_key=six.text_type(course_key)
+        ).first()
+
+        last_this_command_was_run = None
+        if course_node:
+            last_this_command_was_run = course_node['time_last_dumped_to_neo4j']
+
+        return last_this_command_was_run
+
+    @staticmethod
+    def get_course_last_published(course_key):
+        """
+        We use the CourseStructure table to get when this course was last
+        published.
+        Args:
+            course_key: a CourseKey
+
+        Returns: None, if there's no record of the last time this course was
+            published, or the datetime the course was last published at,
+            converted into text.
+        """
+        try:
+            structure = CourseStructure.objects.get(course_id=course_key)
+            course_last_published_date = six.text_type(structure.modified)
+        except CourseStructure.DoesNotExist:
+            course_last_published_date = None
+
+        return course_last_published_date
+
+    def should_dump_course(self, course_key, graph):
         """
         Only dump the course if it's been changed since the last time it's been
         dumped.
-        :param course_key: a CourseKey object.
-        :return: bool. Whether or not this course should be dumped to neo4j.
+        Args:
+            course_key: a CourseKey object.
+            graph: a py2neo Graph object.
+
+        Returns: bool of whether this course should be dumped to neo4j.
         """
 
-        last_this_command_was_run = COMMAND_LAST_RUN_CACHE.get(course_key)
-        last_course_had_published_event = COURSE_LAST_PUBLISHED_CACHE.get(
-            course_key
-        )
+        last_this_command_was_run = self.get_command_last_run(course_key, graph)
+
+        course_last_published_date = self.get_course_last_published(course_key)
 
-        # if we have no record of this course being serialized, serialize it
+        # if we don't have a record of the last time this command was run,
+        # we should serialize the course
```

Comment:
If I read through the doc-string it says,  `returns bool of whether this course should be dumped to neo4j.`
So I assume returning `True` here means we should dump the course, right?

If so, the following comment unclear. Thoughts? 
` # we should serialize the course`